### PR TITLE
fivetran-destination: Add a logging layer

### PR DESCRIPTION
**Pull request by ParkMyCar**
This PR adds a type that implements [`FormatEvent`](https://docs.rs/tracing-subscriber/latest/tracing_subscriber/fmt/trait.FormatEvent.html) that conforms to the [Fivetran SDK logging format](https://github.com/fivetran/fivetran_sdk/blob/main/development-guide.md#logging)

Note: this doesn't add any additional logging yet, just implements the format and enables it for the destination.

### Motivation

Fixes https://github.com/MaterializeInc/materialize/issues/24445

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [x] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  - N/a

### CODE REVIEW

**Review by jkosh44 — APPROVED** [review]
LGTM!

File: src/fivetran-destination/src/logging.rs [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+    N: for<'a> FormatFields<'a> + 'static,
+{
+    fn format_event(
+        &self,
+        ctx: &FmtContext<'_, S, N>,
+        mut writer: format::Writer<'_>,
+        event: &Event<'_>,
+    ) -> std::fmt::Result {
+        let mut message = String::new();
+        ctx.format_fields(Writer::new(&mut message), event)?;
```
Comment (guswynn) [inline src/fivetran-destination/src/logging.rs:48]: what does this do?

**Review by guswynn — APPROVED** [review]
This looks great! 

might be open-sourceable at some point!

File: src/fivetran-destination/src/logging.rs (hunk 1 above)
Comment (ParkMyCar) [reply src/fivetran-destination/src/logging.rs:48]: AFAIU a single event doesn't necessarily contain all of the "fields" it has associated with it, and you need to use a visitor pattern to properly format them. Using `format_fields` from the `FmtContext` does this.

It's also what the implements of `FormatEvent` from `tracing_subscriber` do, [example](https://github.com/tokio-rs/tracing/blob/master/tracing-subscriber/src/fmt/format/mod.rs#L1034)

**Comment by ParkMyCar** [comment]
> This looks great!
> 
> might be open-sourceable at some point!

Thanks! I was thinking the same thing!

**Pull request merged**